eox_nelp/signals/receivers.py

```diff
@@ -266,3 +273,31 @@ def update_async_tracker_context(sender, *args, **kwargs):  # pylint: disable=un
     tracker_context = request.get("kwargs", {}).pop("tracker_context", {})
     current_tracker = tracker.get_tracker()
     current_tracker.enter_context("asynchronous_context", tracker_context)
+
+
+def emit_subsection_attempt_event(usage_id, user_id, *args, **kwargs):  # pylint: disable=unused-argument
+    """This emits  the 'nelc.eox_nelp.grades.subsection.submitted' event
+    when a graded subsection has been attempted.
+    """
+    store = modulestore()
+    user = User.objects.get(id=user_id)
+    usage_key = UsageKey.from_string(usage_id)
+    vertical = store.get_item(store.get_parent_location(usage_key))
+    subsection = vertical.get_parent()
+    course = store.get_course(usage_key.course_key)
+    subsection_grade_factory = SubsectionGradeFactory(user, course=course)
+    subsection_grade = subsection_grade_factory.create(subsection=subsection, read_only=True, force_calculate=True)
+
+    if subsection_grade.graded:
```

Comment:
Is related that a user attempts a subsection, then the subsection is always graded. Or does it depend on the subsection that needs to be configured with grade policies?

Reply:
the subsection must be set with a grading policy